### DISCUSSION TITLE
Accessing variables within getServerData() to return comments for page id within a template

### DISCUSSION THREAD

**Discussion by andrewdever** [gatsbyjs/gatsby#34723] [Help]
I'm trying to use SSR (getServerData() in Gatsby 4) in a template page, to retrieve _just_ the latest comments on a template page that has a graphql query pulling all other data at buildtime.

1. Is this possible? And is it the best approach to pulling just new comments on the server while pulling everything else on build?
2. How do I pass/access variables (the $id of the page) into getServerData()?

**Comment by rsteinmann** [comment]
So I am still investigating this currently but I have good news so far :)

I was able to run the `getServerData` function in a template-page. I was even able to to access the `pageContext` that gets passed by the gatsby-node.js when it fires the `createPage` function. So you can theoretically pass in any data at build time. I am still playing with it though and haven't tested this on production yet.

Next step I'll try to access an external graphQL endpoint to fetch data and pass it back to the page. I'll let you know if this worked out.

**Reply by rsteinmann** [reply]
Okay I was able to successfully use graphQL. Though for my purpose it was enough to use the native javascript fetch method for this as mentioned here: https://graphql.org/graphql-js/graphql-clients/

I am pretty sure you can also use any other package/client since the module exporting works just like a charm. It'll bundle it for you for the node environment I guess.

So yea Gatsby is awesome. Just the docs suck xD
